fees/betswirl/index.ts

```diff
@@ -0,0 +1,212 @@
+      for (const token of graphRes.tokens) {
+        const tokenKey = chain + `:` + token.id;
+
+        totalUserFees[tokenKey] = fromWei(
+          toBN(token.dividendAmount)
+            .plus(token.bankAmount)
+            .plus(token.partnerAmount)
+            .plus(token.treasuryAmount)
+            .plus(token.teamAmount)
+        ).toNumber();
+        totalFees[tokenKey] = totalUserFees[tokenKey];
+
+        totalSupplySideRevenue[tokenKey] = fromWei(
+          toBN(token.bankAmount).plus(token.partnerAmount)
+        ).toNumber();
+
+        totalProtocolRevenue[tokenKey] = fromWei(
+          toBN(token.treasuryAmount).plus(token.teamAmount)
+        ).toNumber();
+        totalDailyHoldersRevenue[tokenKey] = fromWei(
+          token.dividendAmount
+        ).toNumber();
+        totalRevenue[tokenKey] =
+          totalProtocolRevenue[tokenKey] + totalDailyHoldersRevenue[tokenKey];
+      }
+
+      for (const token of graphRes.yesterdayTokens) {
+        const tokenKey = chain + `:` + token.id;
+
+        dailyUserFees[tokenKey] =
+          totalUserFees[tokenKey] -
+          fromWei(
+            toBN(token.dividendAmount)
+              .plus(token.bankAmount)
+              .plus(token.partnerAmount)
+              .plus(token.treasuryAmount)
+              .plus(token.teamAmount)
+          ).toNumber();
+        dailyFees[tokenKey] = dailyUserFees[tokenKey];
+
+        dailyHoldersRevenue[tokenKey] =
+          totalDailyHoldersRevenue[tokenKey] -
+          fromWei(token.dividendAmount).toNumber();
+        dailyProtocolRevenue[tokenKey] =
+          totalProtocolRevenue[tokenKey] -
+          fromWei(toBN(token.treasuryAmount).plus(token.teamAmount)).toNumber();
+        dailyRevenue[tokenKey] =
+          dailyHoldersRevenue[tokenKey] + dailyProtocolRevenue[tokenKey];
+
+        dailySupplySideRevenue[tokenKey] =
+          totalSupplySideRevenue[tokenKey] -
+          fromWei(toBN(token.bankAmount).plus(token.partnerAmount)).toNumber();
+      }
+
+      return {
+        timestamp,
+        dailyFees,
+        dailyUserFees,
+        dailyRevenue,
+        dailyProtocolRevenue,
+        dailyHoldersRevenue,
+        dailySupplySideRevenue,
+        totalFees,
+        totalUserFees,
+        totalRevenue,
+        totalProtocolRevenue,
+        totalSupplySideRevenue,
+        totalDailyHoldersRevenue,
+      };
+    };
+  };
+}
+
+const meta = {
+  methodology: {
+    UserFees:
+      "The player is charged of the fee when a bet is won.",
+    Fees: "All fees (called «house edge» from 2.4% to 3.5% of the payout) comes from the player's bet. The fee has several allocations: Bank, Partner, Dividends, Treasury, and Team.",
+    Revenue: "Dividends, Treasury and Team fee allocations.",
+    ProtocolRevenue: "Treasury and Team fee allocations.",
+    HoldersRevenue: "Dividends fee allocations.",
+    SupplySideRevenue: "Bank and Partner fee allocations",
+  },
+  hallmarks: [
```

Comment:
hallmarks NOT support on this adapter if you want to add this, you should add this repo 
https://github.com/DefiLlama/DefiLlama-Adapters

Reply:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/81955498/217850767-94454c80-4dff-4549-96bf-04719a096b0e.png">
https://docs.llama.fi/list-your-project/other-dashboards/how-to-build-an-adapter#baseadapter

Is is written in the documentation, are you sure?

Reply:
sorry for confusing, from doc there is two places, but if sure I advise you add on tvl https://github.com/DefiLlama/DefiLlama-Adapters 🙏